sql/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -201,18 +201,27 @@ private void advanceToNextFileInput() throws IOException {
         Predicate<URI> uriPredicate = generateUriPredicate(fileInput, fileUri.globPredicate);
         List<URI> uris = getUris(fileInput, fileUri.uri, fileUri.preGlobUri, uriPredicate);
 
-        if (uris.size() > 0) {
+        if (uris.isEmpty()) {
+            LOGGER.warn("No files matched the pattern '{}'", fileUri.uri);
+        } else {
             currentInputIterator = uris.iterator();
             advanceToNextUri(fileInput);
         }
     }
 
     private void initCurrentReader(FileInput fileInput, URI uri) throws IOException {
-        InputStream stream = fileInput.getStream(uri);
-        if (stream != null) {
-            currentReader = createBufferedReader(stream);
-            currentLineNumber = 0;
-            lineProcessor.readFirstLine(currentUri, inputFormat, currentReader);
+        try {
+            InputStream stream = fileInput.getStream(uri);
+            if (stream != null) {
```

Comment:
afaik the stream cannot be null anymore with your change

Reply:
right, S3Input still returns null. can you please mark the `getStream` method as Nullable?